client/ayon_resolve/plugins/create/create_editorial_package.py

```diff
@@ -1,88 +1,120 @@
 
+from ayon_resolve.api import lib, constants
+from ayon_resolve.api.plugin import ResolveCreator, get_editorial_publish_data
 
-class CreateEditorialPackage(LegacyCreator):
+
+class CreateEditorialPackage(ResolveCreator):
     """Create Editorial Package."""
 
-    name = "editorial_pkg"
+    identifier = "io.ayon.creators.resolve.editorial_pkg"
+    product_name = "editorial_pkgMain"
     label = "Editorial Package"
     product_type = "editorial_pkg"
     icon = "camera"
     defaults = ["Main"]
 
-    def process(self):
-        """Process the creation of the editorial package."""
-        project_name = get_current_project_name()
-        folder_path = self.data["folderPath"]
-
-        current_folder = get_folder_by_path(project_name, folder_path)
-        print(current_folder)
+    def create(self, subset_name, instance_data, pre_create_data):
+        """
+        """
+        super(CreateEditorialPackage, self).create(subset_name,
+                                           instance_data,
+                                           pre_create_data)
 
         current_timeline = lib.get_current_timeline()
 
-        context = ask_for_context(
-            project_name, current_folder["id"]
-        )
-
-        if context is None:
-            return
-
-        # Get workfile path to save to.
-        project_name = context["project_name"]
-        folder = get_folder_by_id(project_name, context["folder_id"])
-
-        # task is optional so we need to check if it is set
-        task = None
-        if "task_id" in context:
-            task = get_task_by_id(project_name, context["task_id"])
-
-        # reset self.data to be pointing in the set context data
-        self.data["folderPath"] = folder["path"]
-
-        if task:
-            self.data.update({
-                "taskId": task["id"],
-                "taskName": task["name"],
-            })
-
         if not current_timeline:
-            raise RuntimeError("Make sure to have an active current timeline.")
+            raise CreatorError("Make sure to have an active current timeline.")
 
         timeline_media_pool_item = lib.get_timeline_media_pool_item(
             current_timeline
         )
 
-        publish_data = deepcopy(self.data)
+        publish_data = deepcopy(instance_data)
+
         # add publish data for streamline publishing
         publish_data["publish"] = get_editorial_publish_data(
-            folder_path=folder["path"],
-            product_name=self.data["productName"],
+            folder_path=instance_data["folderPath"],
+            product_name=self.product_name,
         )
 
+        publish_data["label"] = current_timeline.GetName()
         timeline_media_pool_item.SetMetadata(
-            lib.pype_tag_name, json.dumps(publish_data)
+            constants.AYON_TAG_NAME, json.dumps(publish_data)
         )
 
+        publish_data["media_pool_item_id"] = timeline_media_pool_item.GetUniqueId()
```

Comment:
Why does it need this stored, if we're also adding the actual item as `transient_data["timeline_item"]`?